src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobMetadataStore.cs

```diff
@@ -130,12 +129,14 @@ public Task<DicomDataset> GetInstanceMetadataAsync(VersionedInstanceIdentifier v
 
         return ExecuteAsync(async t =>
         {
-            await using (Stream stream = _recyclableMemoryStreamManager.GetStream(GetInstanceMetadataStreamTagName))
-            {
-                await blobClient.DownloadToAsync(stream, cancellationToken);
+            BlobDownloadResult result = await blobClient.DownloadContentAsync(t);
 
+            // DICOM metadata file was saved using UTF-8 encoding with BOM, When using JsonSerializer, by design UTF-8 BOM should be ignored
+            // Else, the following exception will occur "System.Text.Json.JsonReaderException: ''0xEF' is an invalid start of a value. LineNumber: 0 | BytePositionInLine: 0." at the line Reader.Read.
+            // So reading the content into stream which will remove the UTF-8 BOM. https://github.com/dotnet/runtime/issues/29838
+            using (Stream stream = _recyclableMemoryStreamManager.GetStream(result.Content))
```

Comment:
We can skip the use of this `MemoryStream`. Instead just pass `result.ToStream()` to the `DeserializeAsync` method. No need to seek it back to the beginning either. This is not a network stream; it is the entire buffered file.

In some future when we don't need the serializer to seek, we can simply use `DownloadStreaming` instead

Reply:
Yeah, it wasn't because of the API. It was deserializing